lib/helpers/isSimpleValidator.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+/*!
+ * Determines if `arg` is a flat object.
+ *
+ * @param {Object|Array|String|Function|RegExp|any} arg
+ * @api private
+ * @return {Boolean}
+ */
+const isSimpleValidatorCache = new WeakMap();
+
+module.exports = function isSimpleValidator(obj) {
```

Comment:
My only request would be to get rid of the caching. I know that it should be faster and the risk of `WeakMap()` causing any sort of memory issue is extremely low, but I'd rather write off that risk altogether and just keep this as a simple loop.